src/ILLink.CodeFix/RequiresAssemblyFilesCodeFixProvider.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Threading.Tasks;
+using ILLink.RoslynAnalyzer;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Editing;
+
+namespace ILLink.CodeFix
+{
+	[ExportCodeFixProvider (LanguageNames.CSharp, Name = nameof (RequiresAssemblyFilesCodeFixProvider)), Shared]
+	public class RequiresAssemblyFilesCodeFixProvider : BaseAttributeCodeFixProvider
+	{
+		private const string s_title = "Add RequiresAssemblyFiles attribute to parent method";
+
+		public sealed override ImmutableArray<string> FixableDiagnosticIds
+			=> ImmutableArray.Create (RequiresAssemblyFilesAnalyzer.IL3000, RequiresAssemblyFilesAnalyzer.IL3001, RequiresAssemblyFilesAnalyzer.IL3002);
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			await BaseRegisterCodeFixesAsync (context, AttributeableParentTargets.Method | AttributeableParentTargets.Property | AttributeableParentTargets.Event, RequiresAssemblyFilesAnalyzer.FullyQualifiedRequiresAssemblyFilesAttribute, s_title);
+		}
+
+		internal override SyntaxNode[] GetAttributeArguments (SemanticModel semanticModel, SyntaxNode targetNode, CSharpSyntaxNode containingDecl, SyntaxGenerator generator, Diagnostic diagnostic)
+		{
+			var containingSymbol = semanticModel.GetDeclaredSymbol (containingDecl);
+			var name = semanticModel.GetSymbolInfo (targetNode).Symbol?.Name;
+			if (string.IsNullOrEmpty (name) || HasPublicAccessibility (containingSymbol!)) {
+				return Array.Empty<SyntaxNode> ();
+			} else {
+				return new[] { generator.AttributeArgument ("Message", generator.LiteralExpression ($"Calls {name}")) };
```

Comment:
Should also add the `Message` argument in the RUC code fixer?

Reply:
This is also included in the refactor code I'm preparing, if you want I can include the changes and make a review of both adding the codefix and refactoring things

Reply:
I'm fine with having another PR for addressing the RUC changes.

Reply:
I just remember that I tried this, so this AttributeArgument will create a named argument something like
```C#
[RequiresUnreferencedCodeAttribute("message")]
    public int M1() => 0;
[RequiresUnreferencedCode(Message = "Calls M1")]
    int M2() => M1();
```
which will end up generating `error CS0617: 'Message' is not a valid named attribute argument.`
We would need to change the RUC definition to have Message as a named optional argument